run.ts

```diff
@@ -0,0 +1,258 @@
+      yield [key, { value, source: "automatic var" }];
+    } else {
+      // Unreachable.
+      throw new Error(`Undefined variable: ${key}`);
+    }
+  }
+}
+
+function loadJsonc(filePath: string) {
+  const jsoncData = fs.readFileSync(filePath, "utf8");
+  const jsonData = jsoncData
+    .replaceAll(/\s*\/\*.*?\*\//gs, "") // remove multi-line comments
+    .replaceAll(/\/\/.*/g, ""); // remove single-line comments
+  return JSON.parse(jsonData);
+}
+
+function substitute(vars: Vars, value: string) {
+  for (const key in vars) {
+    let re = new RegExp(`\\$(${key}\\b|\\{\\s*${key}\\s*\\})`, "g");
+    // JavaScript doesn't allow lazy substitutions, so we check if
+    // the substitution needs to be done first.
+    if (value.match(re)) {
+      // Substitute recursively to handle nested substitutions.
+      // Since JS doesn't do lazy evaluation, this is technically
+      // O(n^2) worst case, but in reality we should only have a
+      // handful of variables with a couple of levels of recursion.
+      // Doing graph traversals or topological sort should not be
+      // needed, unless we get into a pathological case.
+      value = value.replaceAll(re, substitute(vars, vars[key]));
+    }
+  }
+  return value;
+}
+
+function uniqueId(length = 6) {
+  const min = 2 ** 32;
+  const max = 2 ** 64;
+  return Math.floor(Math.random() * max + min)
+    .toString(36)
+    .slice(0, length);
+}
+
+function defaultProject(): string {
+  const cmd = "gcloud config get-value project";
+  return execSync(cmd).toString().trim();
+}
+
+function accessSecret(secretPath: string): string {
+  const [projectId, ...secretIdParts] = secretPath.split("/");
+  const secretId = secretIdParts.join("/");
+  const cmd = `gcloud --project=${projectId} secrets versions access "latest" --secret=${secretId}`;
+  return execSync(cmd).toString();
+}
+
+function getIdToken(projectId: string): string {
+  const cmd = `gcloud --project=${projectId} auth print-identity-token`;
+  return execSync(cmd).toString().trim();
+}
+
+const command = process.argv[2];
+switch (command) {
+  case "lint": {
+    const usageLint = usage("lint <package-path>");
+    const packagePath = process.argv[3];
+    if (!packagePath) {
+      console.error("Please provide the package path to lint.");
+      console.error(usageLint);
+      process.exit(1);
+    }
+    lint(packagePath);
+    break;
+  }
+
+  case "test": {
+    const usageTest = usage("test <config-path> <package-path>");
+    const configPath = process.argv[3];
+    if (!configPath) {
+      console.error("Please provide the config file path.");
+      console.error(usageTest);
+      process.exit(1);
+    }
+    const packagePath = process.argv[4];
+    if (!packagePath) {
+      console.error("Please provide the package path to lint.");
+      console.error(usageTest);
+      process.exit(1);
+    }
+    test(configPath, packagePath);
+    break;
+  }
+
+  default: {
+    console.error(`Unknown command: ${command}`);
```

Comment:
If you execute `node run.ts` without args you get

```
Unknown command: undefined
usage: node run.ts [lint | test] [options]
```
The response shouldn't return "undefined", just the usage (to match the feel of other script executions, like bash)